packages/shorebird_cli/lib/src/commands/patch/patch_aar_command.dart

```diff
@@ -206,18 +206,24 @@ Please re-run the release command for this version or create a new release.''');
       unzipFn: _unzipFn,
     );
 
-    final shouldContinue =
-        await patchDiffChecker.confirmUnpatchableDiffsIfNecessary(
-      localArtifact: File(
-        aarArtifactPath(
-          packageName: shorebirdEnv.androidPackageName!,
-          buildNumber: buildNumber,
+    final bool shouldContinue;
+    try {
+      shouldContinue =
+          await patchDiffChecker.confirmUnpatchableDiffsIfNecessary(
+        localArtifact: File(
+          aarArtifactPath(
+            packageName: shorebirdEnv.androidPackageName!,
+            buildNumber: buildNumber,
+          ),
         ),
-      ),
-      releaseArtifactUrl: Uri.parse(releaseAarArtifact.url),
-      archiveDiffer: _archiveDiffer,
-      force: force,
-    );
+        releaseArtifactUrl: Uri.parse(releaseAarArtifact.url),
+        archiveDiffer: _archiveDiffer,
+        force: force,
+      );
+    } on UnpatchableChangeException {
```

Comment:
I'm trying to think how I would do this in Rust.

I suspect what you may end up wanting is for both the normal exit and the failed exit to both be handled via exception and whatever caller to this be able to handle such an exception.

Then callers just look like this:

```
await patchDiffChecker.confirmUnpatchableDiffsIfNecessary();
```

And some outer layer knows about:

try {
command()
} on ExitException catch (e) {
  return e.exitCode;
}

Or similar?

Reply:
I like that. Just pushed a commit to standardize the control flow.